docs/advanced.md

```diff
@@ -22,6 +22,23 @@ For advanced customization of this file you can create another file called `conf
 
 The `config.toml.tmpl` will be treated as a Go template file, and the `config.Node` structure is being passed to the template. See [this template](https://github.com/k3s-io/k3s/blob/master/pkg/agent/templates/templates.go#L16-L32) for an example of how to use the structure to customize the configuration file.
 
+## Configuring an HTTP proxy for containerd
+
+If you are running RKE2 in an environment, which only has external connectivity through an HTTP proxy, you can configure your proxy settings on the RKE2 systemd service. These proxy settings will then be passed down to the embedded containerd, so that container images are pulled through the proxy.
+
+Add the necessary `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` variables to the environment file of your systemd service, usually:
+
+- `/etc/default/rke2-server`
+- `/etc/default/rke2-agent`
+
+The `NO_PROXY` variable must include your internal networks, as well as the cluster pod and service IP ranges.
+
+```
+HTTP_PROXY=http://your-proxy.example.com:8888
+HTTPS_PROXY=http://your-proxy.example.com:8888
+NO_PROXY=127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local
+```
+
```

Comment:
You can also prefix environment variables with CONTAINERD_ and they will be passed through to containerd only, with the prefix removed. If your intent is to set the proxy variables just for containerd, that's probably a better way to do it. Would you mind adding this to the documentation as well? The currently recommended variables will set it for both RKE2, the Kubelet, and containerd.

Reply:
I added a section for this.